### PR TITLE
Clean up namespaces for dead trees elimination

### PR DESCRIPTION
**Pull request by xliang6**
Since optimization classes use dynamic inheritance with virtual methods
by default, there is no need to create an optimization class under OMR
namespace and then map to TR namespace.

Signed-off-by: Xiaoli Liang <xsliang@ca.ibm.com>

### CODE REVIEW

**Comment by xliang6** [comment]
@0xdaryl Please review.

File: compiler/optimizer/DeadTreesElimination.hpp [hunk 1]
```diff
@@ -16,22 +16,85 @@
  *    Multiple authors (IBM Corp.) - initial implementation and documentation
  *******************************************************************************/
 
-#ifndef TR_DEADTREESELIMINATION_INCL
-#define TR_DEADTREESELIMINATION_INCL
+#ifndef TR_DEADTREESELIMATION_INCL
+#define TR_DEADTREESELIMATION_INCL
```
Comment (0xdaryl) [inline compiler/optimizer/DeadTreesElimination.hpp]: Why were these renamed to an incorrect spelling of "ELIMINATION" ?

Comment (xliang6) [inline compiler/optimizer/DeadTreesElimination.hpp]: Fixed now. Thanks for catching this -- It was a typo.

**Comment by xliang6** [comment]
@0xdaryl Please review.
